home-assistant-plugin/custom_components/xknx/__init__.py

```diff
@@ -148,11 +148,13 @@ async def async_setup(hass, config):
                 create_knx_device(
                     hass, platform, hass.data[DATA_XKNX].xknx, device_config
                 )
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass, platform.value, DOMAIN, {}, config
-                )
+
+    for platform in SUPPORTED_PLATFORMS:
+        hass.async_create_task(
+            discovery.async_load_platform(
+                hass, platform.value, DOMAIN, {}, config
             )
+        )
```

Comment:
should we filter for only configured platforms? or is it a negligible overhead to just load all platforms even if there are no devices? 

It does create log entries for all supported platforms, even if not configured. Eg. for cover:
> [homeassistant.loader] Loaded cover from homeassistant.components.cover
> [homeassistant.setup] Setting up cover
> [homeassistant.setup] Setup of domain cover took 0.0 seconds
> [homeassistant.components.cover] Setting up cover.xknx

Reply:
Since it's all running async anyway there is no performance downside to it. I'd just leave it as it is.